test/turing.jl

```diff
@@ -0,0 +1,25 @@
+@testset "turing compat" begin
+    k = SqExponentialKernel()
+    y = randn(3)
+    X = randn(3, 1)
+    x = [rand(1) for _ in 1:3]
+    @model GPRegression(y, X) = begin
+        # Priors.
+        α ~ LogNormal(0.0, 0.1)
+        ρ ~ LogNormal(0.0, 1.0)
+        σ² ~ LogNormal(0.0, 1.0)
+
+        # Realized covariance function
+        kernel = α * transform(SqExponentialKernel(), 1/ρ)
+        f = GP(kernel)
+
+        # Sampling Distribution.
+        y ~ f(X, σ²)
+    end
+    # Test for matrices
+    m = GPRegression(y, RowVecs(X))
+    @test_nowarn sample(m, HMC(0.5, 1), 5)
```

Comment:
Wasn't this erroring before? The problem with `@test_nowarn` is that AdvancedHMC is usually quite verbose and prints many warning messages about non-finite gradients etc. So I don't think we should use it here. Maybe rather just use
```suggestion
    sample(m, HMC(0.5, 1), 5)
```
(since tests would fail if it errors) or maybe check something like
```suggestion
    @test length(sample(m, HMC(0.5, 1), 5)) == 5
```